frappe/auth.py

```diff
@@ -209,7 +209,15 @@ def authenticate(self, user = None, pwd = None):
 			self.fail(_('Incomplete login details'), user=user)
 
 		# Ignore password check if tmp_id is set, 2FA takes care of authentication.
-		validate_password = not bool(frappe.form_dict.get('tmp_id'))
+		tmp_id = bool(frappe.form_dict.get('tmp_id'))
+
+		# Check if 2FA is enabled or not
+		enabled = int(frappe.db.get_value('System Settings', None, 'enable_two_factor_auth') or 0)
+
+		if tmp_id and enabled:
+			validate_password = False
+		
+		validate_password = True
```

Comment:
This is a bit confusing. Why check all that in the previous lines if you want to set `validate_password` to always be `True`?

Reply:
That was a mistake !! Thanks. I updated it